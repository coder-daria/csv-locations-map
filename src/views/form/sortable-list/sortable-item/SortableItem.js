import React, { memo } from 'react';
import { string, bool } from 'prop-types';

import { SortableElement } from 'react-sortable-hoc';

import { Label, Container } from './SortableItem.styles';

const SortableItem = SortableElement(({ name, isSorting }) => (
  <Container isSorting={isSorting}>
    <Label>{name}</Label>
  </Container>
));

SortableItem.propTypes = {
  name: string.isRequired,
  isSorting: bool.isRequired,
};

export default memo(SortableItem);

