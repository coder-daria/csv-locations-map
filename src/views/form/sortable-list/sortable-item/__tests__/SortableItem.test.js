import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { SortableContainer } from 'react-sortable-hoc';

import SortableItem from '../SortableItem';

const setup = () => {
  const props = {
    name: 'city',
    isSorting: false,
  };

  return shallow(
  <SortableContainer>
    <SortableItem {...props} />, 
  </SortableContainer>,
  { disableLifecycleMethods: true });
};

describe('<SortableItem />', () => {
  it('should match the sanpshot', () => {
    const wrapper = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
