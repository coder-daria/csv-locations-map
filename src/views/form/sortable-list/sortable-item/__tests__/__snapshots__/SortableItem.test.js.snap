// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortableItem /> should match the sanpshot 1`] = `<Component />`;
